compiler/passes/scopeResolve.cpp

```diff
@@ -344,7 +344,10 @@ static void processImportExprs() {
       // A ModuleSymbol's modUseList is intended to be for initializing modules
       // in an appropriate order.  If we're using an enum symbol, we don't
       // need to worry about this.
-      enclosingModule->moduleUseAdd(mod);
+
+      // Avoid a self-reference in the mod-use list
+      if (mod != enclosingModule)
+        enclosingModule->moduleUseAdd(mod);
```

Comment:
Duoh, you're right of course.  I was so excited that I thought of both failure modes that
I overlooked the correct way to solve it.   I am going to do as you propose and resubmit